content/webapp/components/RelatedWorks.tsx

```diff
@@ -15,50 +25,177 @@ type Props = {
   work: Work;
 };
 
+// Returns the century range for a string containing exactly four digits
+const getCenturyRange = (
+  str?: string
+): { tabLabel: string; from: string; to: string } | null => {
+  const match = str?.match(/^(\d{4})$/);
+  if (match) {
+    const year = parseInt(match[0], 10);
+    const centuryStart = Math.floor(year / 100) * 100;
+    const centuryEnd = centuryStart + 99;
+    return {
+      tabLabel: `From ${centuryStart}s`,
+      from: `${centuryStart}-01-01`,
+      to: `${centuryEnd}-12-31`,
+    };
+  }
+  return null;
+};
+
+const fetchRelated = async ({
+  serverData,
+  params,
+  setRelated,
+  work,
+}: {
+  serverData: SimplifiedServerData;
+  params: { [key: string]: string | string[] };
+  setRelated: (results: WorkBasic[]) => void;
+  work: Work;
+}): Promise<void> => {
+  const response = await catalogueQuery('works', {
+    toggles: serverData.toggles,
+    pageSize: 4, // In case we get the current work back, we will still have 3 to show
+    params: {
+      ...params,
+      include: ['production', 'contributors'],
+    },
+  });
+  if (response.type === 'ResultList') {
+    setRelated(
+      response.results
+        .filter(result => result.id !== work.id)
+        .slice(0, 3)
+        .map(toWorkBasic)
+    );
+  }
+};
+
+// Returns a config object for tabs: one per subject label, plus date-range if present
+function getRelatedTabConfig({
+  work,
+  relatedWorks,
+  setRelatedWorks,
+}: {
+  work: Work;
+  relatedWorks: { [key: string]: WorkBasic[] | undefined };
+  setRelatedWorks: Dispatch<
+    SetStateAction<{ [key: string]: WorkBasic[] | undefined }>
+  >;
+}) {
+  const subjectLabels = work.subjects.map(subject => subject.label).slice(0, 3);
+  const dateRange = getCenturyRange(work.production[0]?.dates[0]?.label);
+
+  const config: {
+    [key: string]: {
+      text: string;
+      params: { [key: string]: string | string[] };
+      related: WorkBasic[] | undefined;
+      setRelated: (results: WorkBasic[]) => void;
+    };
+  } = {};
+
+  subjectLabels.forEach(label => {
+    const id = label.replace(/[^a-zA-Z0-9]/g, '-');
```

Comment:
We have a `dasherize` helper in utils/grammer maybe it could be used?

Reply:
Closing as that's done in https://github.com/wellcomecollection/wellcomecollection.org/pull/11937